doc/api/stream.md

```diff
@@ -1830,6 +1830,54 @@ for await (const result of dnsResults) {
 }
 ```
 
+### `readable.forEach(fn[, options])`
+
+<!-- YAML
+added: REPLACEME
+-->
+
+> Stability: 1 - Experimental
+
+* `fn` {Function|AsyncFunction} a function to call on each item of the stream.
+  * `data` {any} a chunk of data from the stream.
+  * `options` {Object}
+    * `signal` {AbortSignal} aborted if the stream is destroyed allowing to
+      abort the `fn` call early.
+* `options` {Object}
+  * `concurrency` {number} the maximal concurrent invocation of `fn` to call
+    on the stream at once. **Default:** `1`.
+  * `signal` {AbortSignal} allows destroying the stream if the signal is
+    aborted.
+* Returns: {Promise} a promise for when the stream has finished.
+
+This method allows iterating a stream easily. For each item in the stream the
+`fn` function will be called. If the `fn` function returns a promise - that
+promise will be `await`ed.
```

Comment:
I would add some info about backpressure here and how it compares with async iteration, on('data') and on('readable').

Reply:
That's actually a really good point because it's like a `for...await` that is essentially push (since you can't pause or backpressure it) but uses the pull mode (listening to `readable` etc).

I think this is better than the alternative (implementing on top of `data`). Any suggestions of how you'd phrase it?

Reply:
I'd also replace the word "easily" as that's not necessarily true. Easier than what? I suggest:

> This method allows iterating a stream with a function.

Which explains the main difference from `for await...of`.

Reply:
@vweevers it's actually a bunch more subtle than that. I would argue the main differences is that forEach takes a concurrency parameter and is push based (it consumes when it is able vs. consumes when the next iteration of the loop arrives).

I can also `break` from inside a for...await but can only abort from inside a `.forEach`.

It's also important that `forEach` still consumes the stream using the pull API (`.on('readable')`) and not the push one (`on('data')`)

Reply:
(I agree "easily" isn't helpful here I just took it from `filter`)